features/config/service/impl/src/main/java/org/opennms/features/config/service/impl/ConfigurationManagerServiceImpl.java

```diff
@@ -34,22 +34,24 @@
 import org.opennms.features.config.dao.api.ConfigSchema;
 import org.opennms.features.config.dao.api.ConfigStoreDao;
 import org.opennms.features.config.dao.impl.util.ValidateUsingConverter;
+import org.opennms.features.config.service.api.ConfigUpdateInfo;
 import org.opennms.features.config.service.api.ConfigurationManagerService;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Component;
 
 import javax.xml.bind.JAXBException;
 import java.io.IOException;
-import java.util.HashSet;
-import java.util.Objects;
-import java.util.Optional;
-import java.util.Set;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.function.Consumer;
 
 @Component
 public class ConfigurationManagerServiceImpl implements ConfigurationManagerService {
     private final static Logger LOG = LoggerFactory.getLogger(ConfigurationManagerServiceImpl.class);
     private final ConfigStoreDao<JSONObject> configStoreDao;
+    // This map contains key: configName value: list of Consumer
+    private final ConcurrentHashMap<String, Collection<Consumer<ConfigUpdateInfo>>> onloadNotifyMap = new ConcurrentHashMap<>();
```

Comment:
I m not clear why the key in the map is the configName (which should be (schemaName btw.) and not ConfigUpdateInfo?
If we change a config isn't is always configName + configId?

Reply:
I am thinking make it safer to trigger reload for all config for under same configName

Reply:
ok, understand. I am good with that.